### ISSUE TITLE
Bug: config function from typescript-eslint doesn't use `ignores` when `extends` is provided.

### ISSUE THREAD

**Issue by Dremora** [typescript-eslint/typescript-eslint#8511]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

typescript-eslint

### Repro Code

I want to apply type-checked rules to most of my codebase with some exceptions. This is the config I have:

### ESLint Config

```TypeScript
tseslint.config({
    files: ['**/*.{ts,tsx}'],
    ignores: ['libs/legacy-lib/**'],
    extends: [
      ...tseslint.configs.strictTypeChecked,
      ...tseslint.configs.stylisticTypeChecked,
    ],
  })
```

### Expected Result

I expect the final config to have `ignores` applied to each of the configs I'm extending from.

### Actual Result

Only `files` are applied, not `ignores`.

### Additional Info

I can see from [the code](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/typescript-eslint/src/config-helper.ts#L94-L100) that `ignores` is not supported. This is somewhat surprising, given that `files` and `ignores` are two sides of the same coin - the ability to provide a list of files which to apply the rules to.

However, adding this functionality will likely be considered a breaking change.

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `typescript-eslint` | `7.0.1` |

**Comment by Flayms** [comment]
You have an workaround for this?

**Comment by auvred** [comment]
> You have any workaround for this?

@Flayms - as a temporary workaround, you can manually map over configs as described in the docs - https://typescript-eslint.io/packages/typescript-eslint#flat-config-extends

Using the example from the original issue, the config can be rewritten as follows:

```ts
[
  ...tseslint.configs.strictTypeChecked,
  ...tseslint.configs.stylisticTypeChecked,
].map(conf => ({
  ...conf,
  files: ['**/*.{ts,tsx}'],
  ignores: ['libs/legacy-lib/**'],
}))
```

However, I hope that the fix (https://github.com/typescript-eslint/typescript-eslint/pull/8567) will be merged this week, so that a new version containing it will be released next Monday.

**Comment by bradzacher** [comment]
This will release next monday [as per our release schedule](https://typescript-eslint.io/users/releases#latest).
Until that time you can use [the canary version](https://typescript-eslint.io/users/releases#canary)
